### DISCUSSION TITLE
Run script when commit/push command finished

### DISCUSSION THREAD

**Discussion by sarmong** [jesseduffield/lazygit#3358] [General]
Is there a way to hook up to specific events, like when code has been commited/pushed?

Usecase:

I have precommit hooks and I want to send a notification when they passed/failed and the `commit` command has finished running (and the terminal is not in focus).

**Comment by mark2185** [comment]
Just for clarification, do the precommit hooks work when you're invoking `git commit/push` from the CLI? Is it the case that they fail to work when you're doing the same actions from `lazygit` or?

**Reply by sarmong** [reply]
That's not what I am saying. I just need a way to run a custom script when `git commit` exited (successfully or not) in lazygit.

**Reply by mark2185** [reply]
Okay, I was just trying to see if you do have something setup with `git` directly because `post-commit` hooks exist, at least according to `man 5 githooks`.

```
   post-commit
       This hook is invoked by git-commit(1). It takes no parameters, and is invoked after a commit is made.

       This hook is meant primarily for notification, and cannot affect the outcome of git commit.
```

Could you please try if you could set a `post-commit` hook that plays a sound or sends a notification and if that gets triggered from within `lazygit`?
